extensions/amp-ad/0.1/amp-ad.js

```diff
@@ -88,6 +99,8 @@ export class AmpAd extends AMP.BaseElement {
       /*OK*/insertAmpExtensionScript(this.getWin(), extensionTag, true);
     }
     copyAttributes(this.element, newChild);
+    newChild.setAttribute('layout', 'fill');
```

Comment:
That constructs the name of the extension tag.  So, networkImplementationTag('adsense') => 'amp-ad-network-adsense-impl'.  It's used both as the tag name and extension name for requesting dymamic extension load.